src/ucs/sys/string.c

```diff
@@ -146,3 +146,20 @@ const char* ucs_sockaddr_str(const struct sockaddr *sock_addr, char *str, size_t
         return "Invalid string";
     }
 }
+
+char* ucs_strncpy_safe(char *dst, const char *src, size_t len)
+{
+    size_t length;
+
+    if (!len) {
+        return dst;
+    }
+
+    length = ucs_min(len, strnlen(src, len) + 1);
```

Comment:
pls add comment that we copy string including null terminator

Reply:
added